docs/grammar/Solidity.g4

```diff
@@ -39,7 +39,7 @@ importAliases: symbol=identifier (As alias=identifier)?;
 /**
  * Path of a file to be imported.
  */
-path: NonEmptyStringLiteral;
```

Comment:
I was not able to understand the underlying problem here. Even if I changed `NonEmptyStringLiteral` to exactly the same rule as defined by `StringLiteral`, the rule was not matching.

Reply:
Maybe @ekpyron could take a look. The main problem with the old grammar was that it did not support the `import` statements, because that `path` rule did never match (even with an non empty string literal, e.g. `import "asd";`).

Reply:
@ekpyron ah this problem was only related to the rule order.

Reply:
Ok, turned out that just changing the rule order created a lot of other parsing problems. I now just removed `NonEmptyStringLiteral` and use for path `StringLiteral`. The problem is that now `syntaxTests/parsing/import_empty.sol` would fail with `test_antlr_grammar.sh`. So I will ignore that test in `test_antlr_grammar.sh`.

Reply:
I just created https://github.com/ethereum/solidity/pull/11229 to properly deal with all of that. If you undo all grammar changes here and just rebase on top of that PR, everything should just work (and it should even test the grammar against the external projects and against external sources, both of which makes a lot of sense I'd say).

Reply:
Great!